.github/workflows/ci.yml

```diff
@@ -59,25 +59,41 @@ jobs:
             with_opencl: false
 
     steps:
-      - uses: actions/checkout@v4
-
+      - uses: actions/checkout@v4        
+      - name: Set up MSVC environment if applicable
+        uses: ilammy/msvc-dev-cmd@v1
+        with:
+          arch: x64
       - uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
           cache: "pip"
-
       - uses: ./.github/actions/setup-system
-
       - name: Install build dependencies
         run: |
-          pip install --upgrade --pre build cython setuptools wheel
-          pip list
+          pip install --upgrade --pre build wheel 
 
-      - name: Build
+      - name: Build Wheel
         env:
           MACOSX_DEPLOYMENT_TARGET: "10.9"
         run: |
-          python -m build --no-isolation
+          echo $PATH
+          cat > mybuild.py <<EOF
+          import os, sys
+          oldpath=os.environ['PATH'].split(':')
+          newpath=[]
+          for p in oldpath:
+              if "Microsoft Visual Studio" in p:
+                  newpath.insert(0, p)
+              else:
+                  newpath.append(p)
+          print(os.linesep.join(newpath))
+          os.environ["PATH"] = ":".join(newpath)
+          
+          from build.__main__ import main
+          main(sys.argv[1:], 'python -m build')
+          EOF
+          python mybuild.py
```

Comment:
We'll have to find a neater way to fix this in a future PR

Reply:
Is this because the tools from MVS are shadoed by the GNU tools as discussed here https://github.com/ilammy/msvc-dev-cmd/issues/25
If you run this in `cmd` instead of `bash` should solve the issue.
```yaml
shell: ${{ runner.os == 'Windows' && 'cmd' || 'bash' }}
```

Reply:
There appears to be an error when using the conditional operator. Sorry about that, I should have checked the code before.

The only solution I found is to split the step in two, like this:

```yaml
      - name: Build Wheel (Windows)
        if: runner.os == 'Windows'
        shell: cmd
        run: |
          python -m build

      - name: Build Wheel (Unix)
        if: runner.os != 'Windows'
        env:
          MACOSX_DEPLOYMENT_TARGET: "10.9"
        shell: bash
        run: |
          python -m build
```

I checked and test pass with this.